roles/jd-client/src/lib/mod.rs

```diff
@@ -54,110 +56,31 @@ pub static IS_NEW_TEMPLATE_HANDLED: AtomicBool = AtomicBool::new(true);
-                            },
-                        }
-                        std::process::exit(0);
-                    }
-                };
-                let task_status: status::Status = task_status.unwrap();
-
-                match task_status.state {
-                    // Should only be sent by the downstream listener
-                    status::State::DownstreamShutdown(err) => {
-                        error!("SHUTDOWN from: {}", err);
-                        tokio::time::sleep(std::time::Duration::from_secs(1)).await;
-                        task_collector
-                            .safe_lock(|s| {
-                                for handle in s {
-                                    handle.abort();
-                                }
-                            })
-                            .unwrap();
-                        tokio::time::sleep(std::time::Duration::from_secs(1)).await;
-                        break;
-                    }
-                    status::State::UpstreamShutdown(err) => {
-                        error!("SHUTDOWN from: {}", err);
-                        tokio::time::sleep(std::time::Duration::from_secs(1)).await;
-                        task_collector
-                            .safe_lock(|s| {
-                                for handle in s {
-                                    handle.abort();
-                                }
-                            })
-                            .unwrap();
-                        tokio::time::sleep(std::time::Duration::from_secs(1)).await;
-                        break;
-                    }
-                    status::State::UpstreamRogue => {
-                        error!("Changin Pool");
-                        tokio::time::sleep(std::time::Duration::from_secs(1)).await;
-                        task_collector
-                            .safe_lock(|s| {
-                                for handle in s {
-                                    handle.abort();
-                                }
-                            })
-                            .unwrap();
-                        upstream_index += 1;
-                        tokio::time::sleep(std::time::Duration::from_secs(1)).await;
-                        break;
-                    }
-                    status::State::Healthy(msg) => {
-                        info!("HEALTHY message: {}", msg);
-                    }
-                }
-            }
-        }
+    pub fn downstream_address(&self) -> SocketAddr {
+        SocketAddr::new(
+            IpAddr::from_str(&self.config.downstream_address).unwrap(),
+            self.config.downstream_port,
+        )
     }
 
-    async fn initialize_jd_as_solo_miner(
+    pub async fn initialize_jd_as_solo_miner(
         &self,
         tx_status: async_channel::Sender<status::Status<'static>>,
-        task_collector: Arc<Mutex<Vec<AbortHandle>>>,
     ) {
         let proxy_config = &self.config;
```

Comment:
`proxy_config` is legacy terminology, we should take this opportunity to rename this to `jdc_config`